### ISSUE TITLE
The picture is not displayed properly in the extension developed based on webview

### ISSUE THREAD

**Issue by zgh0711** [microsoft/vscode#124246]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.56.2 (user setup)
- OS Version: Windows_NT x64 10.0.19042

I have an extension for a CustomTextEditor which runs on webview. After vscode was upgraded to version 1.56, all the pictures were not displayed when I first entered the editor. But after turning off the editor and reopening it, the picture can be displayed again.

In order to facilitate debugging and reproduce the problem, I made a minimum debuggable program, please help me see what the problem is and how to solve it.

In the extension, I pass the `vscode uri` path of a local image to the webview, and the webview will assign this path to the img tag. If everything is normal, the image will be displayed on the webview.

Through debugging, it can be found that the webview received the image path and successfully assigned the img tag, but the performance on the interface did not match the expectations. The specific problems are as follows:
- Click the json file, when the extension is activated for the first time, the image path is obtained, but the image is not displayed.
- Close the json file and reopen it, the picture can be displayed normally.
- Click the `Update img` button on the page, the image path is updated, but the image is not changed.

What it looks like when you first enter:
![first](https://user-images.githubusercontent.com/15401997/119324511-7e0e5080-bcb2-11eb-91cd-5e9a49b639c8.png)

As expected:
![second](https://user-images.githubusercontent.com/15401997/119324530-823a6e00-bcb2-11eb-85df-26d9dd0f1359.png)

The source code of the program is attached below, and the sample files are in the exampleFiles folder.

Tips:Remember to use `npm install` to install dependencies before running. If necessary, please run `npm install` under the folder of `web-project`
[custom-editor-sample.zip](https://github.com/microsoft/vscode/files/6531141/custom-editor-sample.zip)

**Comment by mjbvz** [comment]
One issue per issue please.

For the first issue, please share a minimal project example that demonstrates the problem

**Comment by zgh0711** [comment]
@mjbvz  I have updated the question as requested, please help me.

**Comment by mjbvz** [comment]
Please share a minimal example extension that demonstrates the problem (i.e. not your full extension)

I can't debug your minified extension code

**Comment by zgh0711** [comment]
@mjbvz
Please take a look at this latest zip file, this is a minimal runnable source code, please install extensions and dependencies under the web-designer folder before running.


[MiniExtension.zip](https://github.com/microsoft/vscode/files/6520389/MiniExtension.zip)

**Comment by zgh0711** [comment]
@mjbvz
I updated the issue description and source code program, please help me to look at this problem

**Comment by zgh0711** [comment]
@mjbvz
I updated the issue description and source code program, please help me to look at this problem

**Comment by mjbvz** [comment]
Thanks. The example seems to work fine for me though in VS Code insiders:

![Screen Shot 2021-05-24 at 3 03 07 PM](https://user-images.githubusercontent.com/12821956/119412734-3632fd80-bca1-11eb-88b6-5ec0be046c7d.png)

Haven't tested on windows yet but did you test with insiders?

**Comment by zgh0711** [comment]
I run the example in VS Code insiders,the problem still exists.And I am running on Windows platform

After testing, I found the following situations:
1. When you first enter the page, the network request status of the image is `pending`, so it cannot be displayed normally.
![pending](https://user-images.githubusercontent.com/15401997/119434000-e9573180-bd49-11eb-8d83-97eff9dfe746.png)

2. Click the `Update img` button to modify the image link, and the network request status of the new image is also `pending`.
![pending2](https://user-images.githubusercontent.com/15401997/119434035-f411c680-bd49-11eb-9888-253623dcae79.png)

3. After closing the page and opening it again, the picture is displayed normally, and you can see that it is actually loaded from the `Service worker`.
![service](https://user-images.githubusercontent.com/15401997/119434054-fbd16b00-bd49-11eb-86e1-b61aa8dd8407.png)

4. Click the `Update img` button in the page that opens again to change the image link. The request status is still `pending`, but the same link has already been requested once in the second article. According to the third article, it is inferred that this It should also be loaded from the `Service worker`, but it is not.Its request status is still `pending`.
![pending3](https://user-images.githubusercontent.com/15401997/119434268-608cc580-bd4a-11eb-8f19-694be9978d0f.png)

**Comment by zgh0711** [comment]
I run the example in VS Code insiders,the problem still exists.And I am running on Windows platform

After testing, I found the following situations:
1. When you first enter the page, the network request status of the image is `pending`, so it cannot be displayed normally.
![pending](https://user-images.githubusercontent.com/15401997/119434000-e9573180-bd49-11eb-8d83-97eff9dfe746.png)

2. Click the `Update img` button to modify the image link, and the network request status of the new image is also `pending`.
![pending2](https://user-images.githubusercontent.com/15401997/119434035-f411c680-bd49-11eb-9888-253623dcae79.png)

3. After closing the page and opening it again, the picture is displayed normally, and you can see that it is actually loaded from the `Service worker`.
![service](https://user-images.githubusercontent.com/15401997/119434054-fbd16b00-bd49-11eb-86e1-b61aa8dd8407.png)

4. Click the `Update img` button in the page that opens again to change the image link. The request status is still `pending`, but the same link has already been requested once in the second article. According to the third article, it is inferred that this It should also be loaded from the `Service worker`, but it is not.Its request status is still `pending`.
![pending3](https://user-images.githubusercontent.com/15401997/119434268-608cc580-bd4a-11eb-8f19-694be9978d0f.png)

**Comment by zhiyun1** [comment]
I also met the same problem on Windows, please help to repeat it on Windows platform.

**Comment by m417z** [comment]
I also experience this issue with dynamically fetched files - script, images and font files. It is not consistent, sometimes there's no problem, but usually there is. Started with v1.56, wasn't an issue with v1.55.2. Also reported 18 days ago here: https://github.com/microsoft/vscode/issues/123284

**Comment by mjbvz** [comment]
The root cause seems to be a framework doing something a bit odd inside the webview, but I've tried to work around that

Please test out the next insiders build and let me know if you still see this issue

**Comment by m417z** [comment]
I can confirm that the insider below fixes the issue. I'm using React, not Vue, by the way.

> Version: 1.57.0-insider
> Commit: 1082913dd012e21cc5f35a06f15c480ca447f67f
> Date: 2021-05-27T06:55:24.593Z (12 hrs ago)

**Comment by zhiyun1** [comment]
In fact, the problem has not been solved. I tested with this version (Version: 1.57.0-insider) and the images still haven't been updated. 

I also tested the  minimal project example above, and it repeats the problem.

Tset result:

![20210604103251](https://user-images.githubusercontent.com/81798000/120737646-0b5a6c00-c521-11eb-8300-c05543ce7a7d.png)

Expected result:

![20210604104806](https://user-images.githubusercontent.com/81798000/120738399-617bdf00-c522-11eb-9960-76baad91e3ea.png)
